### PR TITLE
Return correct status codes in departures backend when handling EIS router responses

### PR DESCRIPTION
**Pull request by marcinkrykowski**
Based on: https://github.com/hmrc/transit-movements-trader-at-destination/pull/154 and https://jira.tools.tax.service.gov.uk/browse/CTDA-803. Needs to wait till John is back.

### CODE REVIEW

File: app/controllers/DeparturesController.scala [hunk 1]
```diff
@@ -75,6 +78,14 @@ class DeparturesController @Inject()(cc: ControllerComponents,
 
                   case SubmissionProcessingResult.SubmissionFailureExternal =>
                     BadGateway
+
+                  case SubmissionProcessingResult.SubmissionFailureRejected(submissionResult: EisSubmissionRejected) =>
+                    submissionResult match {
+                      case ErrorInPayload =>
+                        Status(submissionResult.httpStatus)(submissionResult.asString)
```
Comment (crystalmark) [inline app/controllers/DeparturesController.scala:85]: Could you add some tests that would cover this case please?

**Review by crystalmark — CHANGES_REQUESTED** [review]

**Comment by jhkldev** [comment]
test this please

**Comment by jhkldev** [comment]
test this please

**Comment by jhkldev** [comment]
test this please

**Review by amaalali — APPROVED** [review]

**Review by RobertLaverick — CHANGES_REQUESTED** [review]
@jhkldev can you please make sure there are tickets to address using the new update endpoint more broadly, even if it is simply a spike ticket so that the declaration of "depreciated" repository endpoints isn't left by the wayside.

Also, whilst I appreciate the technical value of the change, and I'll let it be merged in, I'd like you to highlight that the spike should obfuscate the database-level considerations from the rest of the system - something that would typically be called DepartureService (we might need to rename our current DepartureService to reclaim that name in the spike - perhaps to ModelBuilder?). Can you please create this ticket and add it to the PR in a reply comment, and then I'll approve.

**Comment by jhkldev** [comment]
> @jhkldev can you please make sure there are tickets to address using the new update endpoint more broadly, even if it is simply a spike ticket so that the declaration of "depreciated" repository endpoints isn't left by the wayside.
> 
> Also, whilst I appreciate the technical value of the change, and I'll let it be merged in, I'd like you to highlight that the spike should obfuscate the database-level considerations from the rest of the system - something that would typically be called DepartureService (we might need to rename our current DepartureService to reclaim that name in the spike - perhaps to ModelBuilder?). Can you please create this ticket and add it to the PR in a reply comment, and then I'll approve.

https://jira.tools.tax.service.gov.uk/browse/CTDA-1025

**Review by RobertLaverick — APPROVED** [review]

**Pull request merged**